prow/plugins/requiresig/requiresig.go

```diff
@@ -120,8 +124,10 @@ func hasSigLabel(labels []github.Label) bool {
 // (6) if the issue has only the sig label, do nothing
 // (7) if the issue has only the needs-sig label, do nothing
 func handle(log *logrus.Entry, ghc githubClient, cp commentPruner, ie *github.IssueEvent, mentionRe *regexp.Regexp) error {
+	// Ignore PRs, closed issues, and events that aren't new issues or sig label
+	// changes.
 	if ie.Issue.IsPullRequest() || ie.Issue.State == "closed" ||
-		(ie.Action != github.IssueActionOpened && ie.Action != github.IssueActionLabeled && ie.Action != github.IssueActionUnlabeled) {
+		(ie.Action != github.IssueActionOpened && !isSigLabelEvent(ie)) {
```

Comment:
This negative logic hurts my head, but ... if this is a `github.IssueActionOpened` we're still going to add the `needs-sig` label, right?

Reply:
sounds right to me.

perhaps we could instead make the sig labeler plugin expose some method like `func ShouldAddSigLabelTo(ie *github.IssueEvent) bool` and use it both here and there 🤔

Reply:
So if that's right does this really fix the bug? This fixes issues where we get label events for a PR but wouldn't we _first_ get the open event, add the `needs-sig`, then later process the `/sig` and label it? Or are we ok with that little bit of churn at the open of each PR?

Reply:
That case is handled by the conditional that follows this one. If the event is an "opened" event we check that the issue body does not have a `/sig` command or sig team mention before proceeding. I'll make some more changes to make this easier to follow.